### PR TITLE
Default Beats Kibana dashboards to KQL instead of Lucene

### PR DESCRIPTION
**Pull request by lukasolson**
Closes https://github.com/elastic/beats/issues/10969.

In Kibana as of 7.0, the default query language is no longer Lucene, but KQL (Kibana Query Language).

This PR updates all of the JSON objects to use the new language, and modifies any existing queries to the new syntax (if needed - most of them stayed the same).

### CODE REVIEW

**Comment by ruflin** [comment]
Thanks for taking care of this. For some reason this change seems to break our "decoding" of the dashboard script. Need to investigate why.

File: filebeat/module/nginx/_meta/kibana/7/dashboard/Filebeat-nginx-overview.json [hunk 1]
```diff
@@ -76,11 +74,8 @@
                         "filter": [], 
                         "index": "filebeat-*", 
                         "query": {
-                            "query_string": {
-                                "analyze_wildcard": true, 
-                                "query": "*"
-                            }
-                        }
+                            "language": "kuery",
+                            "query": ""
```
Comment (ruflin) [inline filebeat/module/nginx/_meta/kibana/7/dashboard/Filebeat-nginx-overview.json]: Looks like here a `}` went missing below.

**Comment by ruflin** [comment]
CI found one more file which needs updating: https://github.com/elastic/beats/blob/master/libbeat/kibana/testdata/beat-6.json

Here is the test failure output:

```
17:39:17         	Error Trace:	index_pattern_generator_test.go:181
17:39:17         	            				index_pattern_generator_test.go:103
17:39:17         	Error:      	Not equal: 
17:39:17         	            	expected: map[string]interface {}{"long":map[string]interface {}{"id":"url", "params":map[string]interface {}{"outputPrecision":5, "urlTemplate":"_a=(query:(language:kuery,query:'context.app.name:\"{{value}}\"'))", "inputFormat":"string", "labelTemplate":"long template", "outputFormat":"float"}}, "alias":map[string]interface {}{"id":"url", "params":map[string]interface {}{"inputFormat":"string", "labelTemplate":"long template", "outputFormat":"float", "outputPrecision":5, "urlTemplate":"_a=(query:(language:kuery,query:'context.app.name:\"{{value}}\"'))"}}}
17:39:17         	            	actual  : map[string]interface {}{"alias":map[string]interface {}{"id":"url", "params":map[string]interface {}{"outputPrecision":5, "urlTemplate":"_a=(query:(language:lucene,query:'context.app.name:\"{{value}}\"'))", "inputFormat":"string", "labelTemplate":"long template", "outputFormat":"float"}}, "long":map[string]interface {}{"id":"url", "params":map[string]interface {}{"outputPrecision":5, "urlTemplate":"_a=(query:(language:lucene,query:'context.app.name:\"{{value}}\"'))", "inputFormat":"string", "labelTemplate":"long template", "outputFormat":"float"}}}
17:39:17         	            	
17:39:17         	            	Diff:
17:39:17         	            	--- Expected
17:39:17         	            	+++ Actual
17:39:17         	            	@@ -8,3 +8,3 @@
17:39:17         	            	    (string) (len=15) "outputPrecision": (float64) 5,
17:39:17         	            	-   (string) (len=11) "urlTemplate": (string) (len=64) "_a=(query:(language:kuery,query:'context.app.name:\"{{value}}\"'))"
17:39:17         	            	+   (string) (len=11) "urlTemplate": (string) (len=65) "_a=(query:(language:lucene,query:'context.app.name:\"{{value}}\"'))"
17:39:17         	            	   }
17:39:17         	            	@@ -18,3 +18,3 @@
17:39:17         	            	    (string) (len=15) "outputPrecision": (float64) 5,
17:39:17         	            	-   (string) (len=11) "urlTemplate": (string) (len=64) "_a=(query:(language:kuery,query:'context.app.name:\"{{value}}\"'))"
17:39:17         	            	+   (string) (len=11) "urlTemplate": (string) (len=65) "_a=(query:(language:lucene,query:'context.app.name:\"{{value}}\"'))"
17:39:17         	            	   }
17:39:17         	Test:       	TestGenerate
17:39:17 FAIL
```

@lukasolson I'm thinking of only getting this into 7.1 instead of 7.0 as it's not directly a bug fix. If I understand it correctly, things will keep working in 7.0 it's just not "optimal"?

**Comment by lukasolson** [comment]
@ruflin That json file has already been updated... In looking at that test, it appears to read something from fields.yml. I wasn't sure if I updated that file correctly... Could you verify?

https://github.com/elastic/beats/pull/11268/files#diff-6cb2068cf82455cd5dbe9bab3e694e53R27

> I'm thinking of only getting this into 7.1 instead of 7.0 as it's not directly a bug fix. If I understand it correctly, things will keep working in 7.0 it's just not "optimal"?

That's technically correct. The change here isn't just to update the objects to the new format, but also to update the query language to KQL instead of Lucene, which is part of our plan to default to KQL in 7.0.

@AlonaNadler @rayafratkina Your thoughts about pushing the default to KQL for Beats dashboards to 7.1?

File: libbeat/kibana/testdata/fields.yml [hunk 2]
```diff
@@ -24,6 +24,8 @@
           value: "_a=(query:(query_string:(analyze_wildcard:!t,query:'error.grouping_key:%22{{value}}%22')))"
         - min_version: 6.0.0
           value: "_a=(query:(language:lucene,query:'context.app.name:\"{{value}}\"'))"
+        - min_version: 7.0.0
```
Comment (ruflin) [inline libbeat/kibana/testdata/fields.yml]: The tests don't support this at the moment, best just change the line above with 6.0.0 to kquery instead and tests should pass. We need to cleanup these tests on our end.

**Comment by ruflin** [comment]
@lukasolson The changes you made on the testing side make sense unfortunately our test suite does not support it. Just change it for 6 even if it's not correct and will clean it up later. To test if it works you can run `go test .` inside `libbeat/kibana` assuming you have go installed and checked out the repo in the right place.

**Comment by ruflin** [comment]
I tested the system dashboard locally and it seems all works as expected. Thanks a lot for doing this change @lukasolson.

This is now in master and will be in 7.x. We should still discuss if we get this into 7.0.

**Pull request merged**